lib/demos/reference/motion_demo_shared_z_axis_transition.dart

```diff
@@ -0,0 +1,241 @@
+import 'package:flutter/material.dart';
+import 'package:animations/animations.dart';
+import 'package:gallery/l10n/gallery_localizations.dart';
+
+// BEGIN sharedZAxisTransitionDemo
+
+class SharedZAxisTransitionDemo extends StatefulWidget {
+  const SharedZAxisTransitionDemo({Key key}) : super(key: key);
+
+  @override
+  _SharedZAxisTransitionDemoState createState() =>
+      _SharedZAxisTransitionDemoState();
+}
+
+class _SharedZAxisTransitionDemoState extends State<SharedZAxisTransitionDemo> {
+  @override
+  Widget build(BuildContext context) {
+    final sharedAxisTransitionBuilder = const SharedAxisPageTransitionsBuilder(
+      transitionType: SharedAxisTransitionType.scaled,
+    );
+
+    return Theme(
+      data: Theme.of(context).copyWith(
+        pageTransitionsTheme: PageTransitionsTheme(builders: {
+          TargetPlatform.android: sharedAxisTransitionBuilder,
+          TargetPlatform.iOS: sharedAxisTransitionBuilder,
+          TargetPlatform.fuchsia: sharedAxisTransitionBuilder,
+          TargetPlatform.linux: sharedAxisTransitionBuilder,
+          TargetPlatform.windows: sharedAxisTransitionBuilder,
+          TargetPlatform.macOS: sharedAxisTransitionBuilder,
+        }),
+      ),
+      child: Navigator(
+        onGenerateRoute: (settings) {
+          return MaterialPageRoute<void>(
+              builder: (context) => Scaffold(
```

Comment:
I recommend not using the `=>` syntax since it makes the formatting weird with multi-line.